nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/tools/command/LatencyDatadogComputation.java

```diff
@@ -0,0 +1,122 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     bdelbosc
+ */
+package org.nuxeo.lib.stream.tools.command;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.coursera.metrics.datadog.model.DatadogGauge;
+import org.coursera.metrics.datadog.transport.HttpTransport;
+import org.coursera.metrics.datadog.transport.Transport;
+import org.nuxeo.lib.stream.codec.Codec;
+import org.nuxeo.lib.stream.computation.ComputationContext;
+import org.nuxeo.lib.stream.computation.Record;
+import org.nuxeo.lib.stream.log.Latency;
+import org.nuxeo.lib.stream.log.LogManager;
+import org.nuxeo.lib.stream.log.internals.LogPartitionGroup;
+
+/**
+ * A computation that sends periodically latencies to Datadog.
+ *
+ * @since 11.1
+ */
+public class LatencyDatadogComputation extends LatencyTrackerComputation {
+
+    private static final Log log = LogFactory.getLog(LatencyDatadogComputation.class);
+
+    protected final String apiKey;
+
+    protected final List<String> tags;
+
+    protected final String basePrefix;
+
+    protected final boolean partition;
+
+    protected HttpTransport transport;
+
+    public LatencyDatadogComputation(LogManager manager, List<String> logNames, String apiKey, List<String> tags,
+            String basePrefix, String computationName, int intervalSecond, int count, boolean partition,
+            boolean verbose, Codec<Record> codec) {
+        super(manager, logNames, computationName, intervalSecond, count, verbose, codec, 0);
+        this.apiKey = apiKey;
+        this.tags = tags;
+        this.basePrefix = basePrefix;
+        this.partition = partition;
+    }
+
+    @Override
+    public void init(ComputationContext context) {
+        super.init(context);
+        transport = new HttpTransport.Builder().withApiKey(apiKey).build();
+    }
+
+    @Override
+    protected void processLatencies(ComputationContext context, LogPartitionGroup logGroup, List<Latency> latencies) {
+        Latency groupLatency = Latency.of(latencies);
+        publishMetrics(groupLatency, basePrefix, "all", logGroup.name, logGroup.group);
+        if (!partition) {
+            return;
+        }
+        for (int part = 0; part < latencies.size(); part++) {
+            publishMetrics(groupLatency, basePrefix, String.format("%02d", part), logGroup.name, logGroup.group);
+        }
+    }
+
+    protected void publishMetrics(Latency latency, String prefix, String partition, String stream, String group) {
+        if (verbose) {
+            log.info(latency.toString());
+        }
+        // upper is the time when the latency has been measured
+        long metricTime = latency.upper() / 1000;
+        String host = "";
```

Comment:
Is it expected to have the empty string here?

Reply:
This is supposed to be the hostname that is sending the metric, but the metrics are global to the cluster, will update this to make it clear.

Reply:
finally, I use the hostname reporting the metric it should not be a problem